### DISCUSSION TITLE
Small non-station grids need a way to generate gravity

### DISCUSSION THREAD

**Discussion by ProfanedBane** [space-wizards/space-station-14#3977] [General]
As it is now any grids that don't have a gravity gen don't have gravity. This sounds fine in theory but it means all shuttles and engineering project mere tiles away from the station have no gravity, this doesn't seem like gud design.

**Comment by ike709** [comment]
Engineers just need magboots.

There should be a mini-gravgen for shuttles, probably.

**Comment by PaulRitter** [comment]
just make a smaller gravgen variant for shuttles 🤷

**Comment by moonheart08** [comment]
Some grids, like asteroids or planets, will need their own gravity without a special object. Should be possible to introduce a marker component for grids that makes them always have gravity. You would need to insert an additional check at [GravitySystem.cs line 51](https://github.com/space-wizards/space-station-14/blob/818bdb3c63b30afa1a884dd03fcca102b298d6cd/Content.Server/GameObjects/EntitySystems/GravitySystem.cs#L51), but that's it and fairly trivial.
A version of the GravityGeneratorComponent without the UI and/or the need for power would also be good. Could simply be some flags on the component. This would allow putting it in sneaky places like making the shuttle control console also generate the shuttle's gravity, or providing gravity using a differently themed mechanism (i.e. a magical floating gravity crystal).

**Comment by ScalyChimp** [comment]
this would also be good for when engineering eventually snaps and decides to fuck off and make it's own station, a mini grav-gen would be needed for something like that.
